pkg/controller/snapshot_controller.go

```diff
@@ -125,9 +125,21 @@ func (ctrl *csiSnapshotController) syncContent(content *crdv1.VolumeSnapshotCont
 		snapshot = nil
 	}
 	if snapshot == nil {
-		ctrl.deleteSnapshotContent(content)
+		if content.Spec.DeletionPolicy != nil {
+			switch *content.Spec.DeletionPolicy {
+			case crdv1.VolumeSnapshotContentRetain:
+				glog.V(4).Infof("VolumeSnapshotContent[%s]: policy is Retain, nothing to do", content.Name)
+		
+			case crdv1.VolumeSnapshotContentDelete:
+				glog.V(4).Infof("VolumeSnapshotContent[%s]: policy is Delete", content.Name)
+				ctrl.deleteSnapshotContent(content)
+			default:
+				// Unknown PersistentVolumeReclaimPolicy
+				ctrl.eventRecorder.Event(content, v1.EventTypeWarning, "SnapshotUnknownDeletionPolicy", "Volume Snapshot Content has unrecognized deletion policy")
```

Comment:
So the default policy is actually the same as Retain?

Reply:
no, it should be Delete. I fixed the logic here.

Reply:
For default, it won't call deleteSnapshotContent.  So default is still equal to Retain?  You said it should be the same as Delete earlier.

Reply:
there are two cases. 
For dynamic creation, we add deletePolicy when creating snapshotContent. Check "createSnapshotOperation" function in controller

For static binding case, if no deletePolicy is set, by default it is retain.